diff --git a/src/core/tcp.c b/src/core/tcp.c
index ea95ffee..15cc8d8f 100644
--- a/src/core/tcp.c
+++ b/src/core/tcp.c
@@ -1784,7 +1789,7 @@ tcp_kill_state(enum tcp_state state)
  * Kills the oldest connection that is in TIME_WAIT state.
  * Called from tcp_alloc() if no more connections are available.
  */
-static void
+void
 tcp_kill_timewait(void)
 {
   struct tcp_pcb *pcb, *inactive;
diff --git a/src/core/tcp_in.c b/src/core/tcp_in.c
index 1b17e40f..61175eb7 100644
--- a/src/core/tcp_in.c
+++ b/src/core/tcp_in.c
@@ -1010,6 +1010,7 @@ tcp_process(struct tcp_pcb *pcb)
           TCP_RMV_ACTIVE(pcb);
           pcb->state = TIME_WAIT;
           TCP_REG(&tcp_tw_pcbs, pcb);
+          TCP_TW_LIMIT(MEMP_NUM_TCP_PCB_TIME_WAIT);
         } else {
           tcp_ack_now(pcb);
           pcb->state = CLOSING;
@@ -1028,6 +1029,7 @@ tcp_process(struct tcp_pcb *pcb)
         TCP_RMV_ACTIVE(pcb);
         pcb->state = TIME_WAIT;
         TCP_REG(&tcp_tw_pcbs, pcb);
+        TCP_TW_LIMIT(MEMP_NUM_TCP_PCB_TIME_WAIT);
       }
       break;
     case CLOSING:
@@ -1038,6 +1040,7 @@ tcp_process(struct tcp_pcb *pcb)
         TCP_RMV_ACTIVE(pcb);
         pcb->state = TIME_WAIT;
         TCP_REG(&tcp_tw_pcbs, pcb);
+        TCP_TW_LIMIT(MEMP_NUM_TCP_PCB_TIME_WAIT);
       }
       break;
     case LAST_ACK:
